engine/api/cache/cache.go

```diff
@@ -128,3 +137,27 @@ func QueueLen(queueName string) int {
 	}
 	return s.QueueLen(queueName)
 }
+
+// Publish a message on a channel
+func Publish(queueName string, value interface{}) {
+	if s == nil {
+		return
+	}
+	s.Publish(queueName, value)
+}
+
+// Subscribe to a channel
+func Subscribe(queueName string) PubSub {
+	if s == nil {
+		return nil
+	}
+	return s.Subscribe(queueName)
+}
+
+// GetMessageFromSubscription Get a message from a subscription
+func GetMessageFromSubscription(pb PubSub, c context.Context) (string, error) {
```

Comment:
context.Context should always be the first arg.
as in the interface: `GetMessageFromSubscription(c context.Context, pb PubSub) `